command_line/stills_process.py

```diff
@@ -332,6 +319,20 @@ def do_work(i, item_list):
         processor = Processor(copy.deepcopy(params), composite_tag = "%04d"%i)
 
         for item in item_list:
+          try:
+            for imageset in item[1].extract_imagesets():
+              update_geometry(imageset)
+          except RuntimeError as e:
+            logger.info("Error updating geometry on item %s, %s"%(str(item[0]), str(e)))
+            continue
+
+          if self.reference_detector is not None:
+            from dxtbx.model import Detector
```

Comment:
suggest moving import from middle of for loop to top of file

Reply:
Wouldn't that create an unneeded import for situation where there is no reference detector?